apps/web/screens/ApiCatalogue/ApiCatalogue.tsx

```diff
@@ -1,30 +1,209 @@
+const LIMIT = 20
+
+const ApiCatalogue: Screen<ApiCatalogueProps> = ({
+  mainContent,
+  staticContent,
+  filterContent,
+}) => {
   const { disableApiCatalog: disablePage } = publicRuntimeConfig
 
   if (disablePage === 'true') {
     throw new CustomNextError(404, 'Not found')
   }
 
-  return <h1>{title}</h1>
+  const sn = useNamespace(staticContent)
+
+  const onLoadMore = () => {
+    if (data?.getApiCatalogue.pageInfo?.nextCursor == null) {
+      return
+    }
+
+    const { nextCursor } = data?.getApiCatalogue?.pageInfo
+    const param = { ...parameters, cursor: nextCursor }
+    fetchMore({
+      variables: { input: param },
+      updateQuery: (prevResult, { fetchMoreResult }) => {
+        fetchMoreResult.getApiCatalogue.services = [
+          ...prevResult.getApiCatalogue.services,
+          ...fetchMoreResult.getApiCatalogue.services,
+        ]
+        return fetchMoreResult
+      },
+    })
+  }
+
+  const [parameters, setParameters] = useState<GetApiCatalogueInput>({
+    cursor: null,
+    limit: LIMIT,
+    query: '',
+    pricing: [],
+    data: [],
+    type: [],
+    access: [],
+  })
+
+  const { data, loading, error, fetchMore, refetch } = useQuery<
+    Query,
+    QueryGetApiCatalogueArgs
+  >(GET_CATALOGUE_QUERY, {
+    variables: {
+      input: parameters,
+    },
+  })
+
+  return (
+    <SubpageLayout
+      main={
+        <SidebarLayout sidebarContent={<div>Navigation menu here</div>}>
+          <SubpageMainContent main={<div>Main content here</div>} />
+        </SidebarLayout>
+      }
+      details={
+        <SubpageDetailsContent
+          header={
+            <Text variant="h4" color="blue600">
+              {sn('title')}
+            </Text>
+          }
+          content={
+            <SidebarLayout sidebarContent={<>List filter here</>}>
+              {(error || data?.getApiCatalogue?.services.length < 1) && (
+                <GridContainer>
+                  {error ? (
+                    <Text>{sn('errorHeading')}</Text>
+                  ) : loading ? (
+                    <LoadingIcon animate color="blue400" size={32} />
+                  ) : (
+                    <Text>{sn('notFound')}</Text>
+                  )}
+                </GridContainer>
+              )}
+              {data?.getApiCatalogue?.services.length > 0 && (
+                <GridContainer>
+                  <ServiceList
+                    services={data?.getApiCatalogue?.services}
+                    tagDisplayNames={filterContent}
+                  />
+                  {data?.getApiCatalogue?.pageInfo?.nextCursor != null && (
+                    <Box display="flex" justifyContent="center">
+                      <Button
+                        colorScheme="default"
+                        iconType="filled"
+                        onClick={() => onLoadMore()}
+                        size="default"
+                        type="button"
+                        variant="ghost"
```

Comment:
no need to define default values
```suggestion
                      <Button
                        onClick={() => onLoadMore()}
                        variant="ghost"
```

Reply:
fixed